File: app/src/main/java/com/prey/actions/geofences/GeofecenceParse.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.geofences;

import android.content.Context;

import com.prey.PreyLogger;
import com.prey.net.PreyWebServices;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class GeofecenceParse {

    public static List<GeofenceDto> getJSONFromUrl(Context ctx) {
        String json = null;
        try {
            json = PreyWebServices.getInstance().geofencing(ctx);
        } catch (Exception e) {
        }
        return getJSONFromTxt(ctx, json);
    }

    public static List<GeofenceDto> getJSONFromTxt(Context ctx, String json) {
        json = "{\"prey\":" + json + "}";
        List<GeofenceDto> listGeofence = new ArrayList<GeofenceDto>();
        PreyLogger.d(json);
        try {
            JSONObject jsnobject = new JSONObject(json);
            JSONArray jsonArray = jsnobject.getJSONArray("prey");
            for (int i = 0; i < jsonArray.length(); i++) {
                String jsonCommand = jsonArray.get(i).toString();
                JSONObject explrObject = new JSONObject(jsonCommand);
                GeofenceDto geofence = new GeofenceDto();
                geofence.id = explrObject.getString("id");
                geofence.name = explrObject.getString("name");
                geofence.latitude = Double.parseDouble(explrObject.getString("lat"));
                geofence.longitude = Double.parseDouble(explrObject.getString("lng"));
                geofence.radius = Float.parseFloat(explrObject.getString("radius"));
                listGeofence.add(geofence);
            }
        } catch (Exception e) {
            return null;
        }
        return listGeofence;
    }
}

File: app/src/main/java/com/prey/actions/geofences/GeofecenceParse.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.geofences

import android.content.Context
import com.prey.PreyLogger
import com.prey.net.PreyWebServices
import org.json.JSONObject

object GeofecenceParse {
    fun getJSONFromUrl(context: Context): List<GeofenceDto>? {
        return try {
            val jsonString = PreyWebServices.getInstance().geofencing(context)
            parseJsonFromString(context, jsonString)
        } catch (e: Exception) {
            PreyLogger.e("Error:" + e.message, e)
            null
        }
    }

    fun parseJsonFromString(context: Context, jsonString: String?): List<GeofenceDto>? {
        val wrappedJsonString = "{\"prey\":$jsonString}"
        val listGeofence: MutableList<GeofenceDto> = ArrayList()
        try {
            val jsnobject = JSONObject(wrappedJsonString)
            val geofenceArray = jsnobject.getJSONArray("prey")
            for (i in 0 until geofenceArray.length()) {
                val geofenceJson = geofenceArray[i].toString()
                val geofenceObject = JSONObject(geofenceJson)
                val geofence = GeofenceDto()
                geofence.setId(geofenceObject.getString("id"))
                geofence.setName(geofenceObject.getString("name"))
                geofence.setLatitude(geofenceObject.getString("lat").toDouble())
                geofence.setLongitude(geofenceObject.getString("lng").toDouble())
                geofence.setRadius(geofenceObject.getString("radius").toFloat())
                listGeofence.add(geofence)
            }
        } catch (e: Exception) {
            return null
        }
        return listGeofence
    }
}